internal/core/metadata/application/device.go

```diff
@@ -75,6 +91,14 @@ func DeleteDeviceByName(name string, ctx context.Context, dic *di.Container) err
 		return errors.NewCommonEdgeXWrapper(err)
 	}
 	go deleteDeviceCallback(ctx, dic, device)
+
+	go func() {
+		if err := publishDeviceSystemEvent(common.DeviceSystemEventActionDelete, device, ctx, lc, dic); err != nil {
+			// Don't fail request if couldn't publish, just log it. Error message already has context.
+			lc.Error(err.Error())
```

Comment:
log could add more context here:
`lc.Errorf("issue happens when publishing the device add system event: %v", err)`

Reply:
No longer needed.